src/chat.rs

```diff
@@ -251,7 +253,7 @@ impl Chat {
             || self.typ == Chattype::VerifiedGroup)
         {
             error!(context, "Cannot send to chat type #{}.", self.typ,);
-            return Ok(0);
+            bail!("Cannot set to chat type #{}", self.typ);
```

Comment:
what's the rationale for changing this?  IIRC we have UIs sometimes doing spurious calls and we do not want to crash.  Haven't checked the caller sites and how they treat an error so bailing out might be ok.

Reply:
One call site is from `prepare_msg_common()` which in turn is called by `send_msg()` and `prepare_msg()`.

In the case of prepare_msg this just assumes the message was prepared and all is fine and dandy, that bubbles up into the C API as a 0 return value though, which means error over there.  So this function was returning an error when called from the FFI but not when called from Rust.  Now it returns an error when called from rust too, and the FFI still sees the error and turns it into the existing 0 return value.

In the case of send_msg it calls this function with a `?`, presumably expecting to error on failures.  However the `Ok(0)` result would have let it carry on.  It would have passed this to a `job_send_msg()?;` call, which would have called `MimeFactory::load_msg` which would have called `Message::load_from_db` and finally we'd be trying to load msg 0, which would fail, all these calls use `?` so this would not do any harm and fail just after a bunch of extra calls instead of a bit earlier, no bug at least.

Finally the other callsite is `forward_msgs()`, there it loaded the message freshly from the database before calling this, which makes all the following a little academic since this msg was already sent in this form so should not fail this check now, anyway.  Again this call uses `?` to bail (perhaps it shouldn't and still forward the other messages, that's what it does in other places but i consider fixing this out of scope here).  The code here does set this invalid result now in some params so who knows what will break later there, one of the branches calls job_send_msg which will fail too so we're back to might as well fail early for that branch.  So while I don't 100% understand the forwarding code I'm pretty certain that this improved things.

So really I think this was a C-style error return.  The current state of the rust conversion does not handle this correctly.  Failing early is the right thing to do here, I believe it was a mistake made when converting this function to Result since from that point on rust callers do no longer expect C-style error reporting.